libs/vkd3d/command.c

```diff
@@ -4045,14 +3876,11 @@ static bool d3d12_command_list_gather_pending_queries(struct d3d12_command_list
         map_buffer.range = entry_buffer_size;
 
         for (j = 0; j < ARRAY_SIZE(vk_writes); j++)
-            vk_writes[j].dstSet = vk_set;
-
-        VK_CALL(vkUpdateDescriptorSets(list->device->vk_device,
-                ARRAY_SIZE(vk_writes), vk_writes, 0, NULL));
+            vk_writes[j].dstSet = VK_NULL_HANDLE;
```

Comment:
This can be moved into the loop above that initializes `vk_writes`, no reason to have this in a nested loop if the set is always null.